fdbserver/workloads/PhysicalShardMove.actor.cpp

```diff
@@ -212,6 +210,41 @@ struct PhysicalShardMoveWorkLoad : TestWorkload {
 		return Void();
 	}
 
+	ACTOR Future<Void> deleteCheckpoints(Database cx, std::vector<UID> checkpointIds) {
+		TraceEvent(SevDebug, "DataMoveDeleteCheckpoints").detail("Checkpoints", describe(checkpointIds));
+
+		state Transaction tr(cx);
+		loop {
+			try {
+				std::vector<Future<Optional<Value>>> checkpointEntries;
+				for (const UID& id : checkpointIds) {
+					checkpointEntries.push_back(tr.get(checkpointKeyFor(id)));
+				}
+				std::vector<Optional<Value>> checkpointValues = wait(getAll(checkpointEntries));
+
+				for (int i = 0; i < checkpointIds.size(); ++i) {
+					const auto& value = checkpointValues[i];
+					if (!value.present()) {
+						TraceEvent(SevWarnAlways, "CheckpointNotFound");
+						continue;
+					}
+					CheckpointMetaData checkpoint = decodeCheckpointValue(value.get());
+					const Key key = checkpointKeyFor(checkpoint.checkpointID);
+					checkpoint.setState(CheckpointMetaData::Deleting);
+					tr.set(key, checkpointValue(checkpoint));
+					tr.clear(singleKeyRange(key));
```

Comment:
It is used for generating a private mutation, so that the SS will delete their local checkpoints. If we use only `clear`, then we have to do a read in the proxies to read all the related SSes, since the SS ids are not encoded into the keys.

Reply:
Nice trick! Can you add a comment about this usage?

Reply:
Done.